mailer.rst

```diff
@@ -0,0 +1,649 @@
+several - install whichever you want:
+
+==================  =============================================
+Service             Install with
+==================  =============================================
+Amazon SES          ``composer require symfony/amazon-mailer``
+MailChimp           ``composer require symfony/mailchimp-mailer``
+Mailgun             ``composer require symfony/mailgun-mailer``
+Postmark            ``composer require symfony/postmark-mailer``
+SendGrid            ``composer require symfony/sendgrid-mailer``
+Gmail               ``composer require symfony/google-mailer``
+==================  =============================================
+
+Each library includes a :ref:`Flex recipe <flex-recipe>` that will add example configuration
+to your ``.env`` file. For example, suppose you want to use SendGrid. First,
+install it:
+
+.. code-block:: terminal
+
+    $ composer require symfony/sendgrid-mailer
+
+You'll now have a new line in your ``.env`` file that you can uncomment:
+
+.. code-block:: bash
+
+    # .env`
+
+    SENDGRID_KEY=
+    MAILER_DSN=smtp://$SENDGRID_KEY@sendgrid
+
+The``MAILER_DSN`` isn't a *real* SMTP address: it's a simple format that offloads
+most of the configuration work to mailer. The ``@sendgrid`` part of the address
+activates the SendGrid mailer library that you just installed, which knows all
+about how to deliver messages to SendGrid.
+
+The *only* part you need to change is to set ``SENDGRID_KEY`` to your key (in
+``.env`` or ``.env.local``).
+
+Each transport will have different environment variables that the library will use
+to configure the *actual* address and authentication for delivery. Some also have
+options that can be configured with query parameters on end of the ``MAILER_DSN`` -
+like ``?region=`` for Amazon SES. Some transports support sending via ``http``
+or ``smtp`` - both work the same, but ``http`` is recommended when available.
+
+Creating & Sending Messages
+---------------------------
+
+To send an email, autowire the mailer using
+:class:`Symfony\\Component\\Mailer\\MailerInterface` (service id ``mailer``)
+and create an :class:`Symfony\\Component\\Mime\\Email` object::
+
+    // src/Controller/MailerController.php
+    namespace App\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\Mailer\MailerInterface;
+    use Symfony\Component\Mime\Email;
+
+    class MailerController extends AbstractController
+    {
+        /**
+         * @Route("/email")
+         */
+        public function sendEmail(MailerInterface $mailer)
+        {
+            $email = (new Email())
+                ->from('hello@example.com')
+                ->to('you@example.com')
+                //->cc('cc@example.com')
+                //->bcc('bcc@example.com')
+                //->replyTo('fabien@example.com')
+                //->priority(Email::PRIORITY_HIGH)
+                ->subject('Time for Symfony Mailer!!!99')
```

Comment:
what is the 99 ?

Reply:
ha! nice catch - a bad copy from some local code when I was building all of this!